src/solvers/refinement/string_refinement.cpp

```diff
@@ -2152,6 +2152,13 @@ static bool is_valid_string_constraint(
   {
     // Condition 1: All indices of the same string must be the of the same form
     const exprt rep=pair.second.back();
+
+    // This test is disabled because Java like "s.startsWith(s, 1)" produces
+    // a constraint like "forall x, s[x] == s[x+1]", which does not meet this
+    // constraint as s is indexed with both 'x' and 'x+1' under the same
+    // universal quantifier. We should either indirect to avoid the double
+    // reference, or establish that this conditon wasn't needed to begin with.
+    /*
```

Comment:
Use `#if 0`

Reply:
No sure I understand "produces a constraint, which does not meet this constraint".

Reply:
I had a think about that. A condition of the PASS algorithm is for the comparison predicate to be non-self-recursive.
Since the algorithm stops on a fix point (i.e. when no more indices are added to the index set), it might happen that adding an index will always result in also adding a shifted index (e.g. in the `s[x] == s[x+1]` case). So the algorithm never reaches a fix point and will enter an infinite loop (in the case of unbounded string lengths).

However, there's hope that early termination will happen anyway because of another mechanism. So in practice that might actually not happen often (or at all).

I would prefer the indirection solution, but if it's too complicated I can live with the pragmatic solution of removing the assertion and seeing what happens.